src/main/java/hudson/scm/IntegritySCM.java

```diff
@@ -899,24 +899,29 @@ private String getSource(EnvVars env) {
   {
     @Extension
     public static final DescriptorImpl INTEGRITY_DESCRIPTOR = new DescriptorImpl();
-    private ConnectionPoolDataSource dataSource;
+    private transient ConnectionPoolDataSource connectionPoolDataSource;

```

Comment:
Do not save going forward, and do not even try to load from an existing XML file if present.